[package] folly/2020.08.10.00: Compile errors on devtoolset-4 gcc5

<!-- 
  Please don't forget to update the issue title.
  Include all applicable information to help us reproduce your problem.
-->

### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **folly/2020.08.10.00**
  * Operating System+version: **Red Hat Enterprise 6**
  * Compiler+version: **GCC 5.3.1 from devtoolset-4**
  * Conan version: **conan 1.29.1**
  * Python version: **Python 3.7.4**


### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
[settings]
os=Linux
os_build=Linux
arch=x86_64
arch_build=x86_64
compiler=gcc
compiler.version=5
compiler.libcxx=libstdc++
build_type=Debug
[options]
[build_requires]
[env]
```


### Steps to reproduce (Include if Applicable)

Build recipe using gcc 5.3.1 (devtoolset-4) on RHEL6.  There are two compile errors:

```
/userhome/epederson/.conan/data/folly/2020.08.10.00/test/test/build/21e92cd0e529b036c562d4dfa4ad1adfee8a5c75/source_subfolder/folly/portability/Time.cpp:278:2: error: #error No clock_gettime(3) compatibility wrapper available for this platform.
 #error No clock_gettime(3) compatibility wrapper available for this platform.
  ^
gmake[2]: *** [source_subfolder/CMakeFiles/folly_base.dir/folly/portability/Time.cpp.o] Error 1
```

```
/userhome/epederson/.conan/data/folly/2020.08.10.00/test/test/build/21e92cd0e529b036c562d4dfa4ad1adfee8a5c75/source_subfolder/folly/Benchmark.cpp:340:30: error: expected ‘)’ before ‘PRId64’
               printf("  %-*" PRId64, int(name.length()), ptr->value);
                              ^
```

I will submit a PR with a fix.

--- a/recipes/folly/all/conandata.yml
+++ b/recipes/folly/all/conandata.yml
@@ -34,3 +34,7 @@ patches:
       base_path: "source_subfolder"
     - patch_file: "patches/0012-compiler-flags.patch"
       base_path: "source_subfolder"
+    - patch_file: "patches/0014-find-librt.patch"
+      base_path: "source_subfolder"
+    - patch_file: "patches/0015-benchmark-format-macros.patch"
+      base_path: "source_subfolder"

--- a/recipes/folly/all/conanfile.py
+++ b/recipes/folly/all/conanfile.py
@@ -179,7 +179,7 @@ class FollyConan(ConanFile):
         if Version(self.version) >= "2020.08.10.00":
             self.cpp_info.components["libfolly"].requires.append("fmt::fmt")
         if self.settings.os == "Linux":
-            self.cpp_info.components["libfolly"].system_libs.extend(["pthread", "dl"])
+            self.cpp_info.components["libfolly"].system_libs.extend(["pthread", "dl", "rt"])
         elif self.settings.os == "Windows":
             self.cpp_info.components["libfolly"].system_libs.extend(["ws2_32", "Iphlpapi", "Crypt32"])
         if (self.settings.os == "Linux" and self.settings.compiler == "clang" and

--- a/recipes/folly/all/patches/0014-find-librt.patch
+++ b/recipes/folly/all/patches/0014-find-librt.patch
@@ -9,7 +9,7 @@ index 6b8b308c7..908d72d51 100644
 +if (CMAKE_SYSTEM_NAME STREQUAL "Linux")
 +  find_library(LIBRT rt)
 +  if (LIBRT)
-+    set(CMAKE_REQUIRED_LIBRARIES "rt")
++    list(APPEND CMAKE_REQUIRED_LIBRARIES "rt")
 +  endif()
 +endif()
 +

--- a/recipes/folly/all/patches/0014-find-librt.patch
+++ b/recipes/folly/all/patches/0014-find-librt.patch
@@ -6,7 +6,7 @@ index 6b8b308c7..908d72d51 100644
    CMAKE_REQUIRED_FLAGS
    "${CMAKE_REQUIRED_FLAGS}")
  
-+if (UNIX)
++if (CMAKE_SYSTEM_NAME STREQUAL "Linux")
 +  find_library(LIBRT rt)
 +  if (LIBRT)
 +    set(CMAKE_REQUIRED_LIBRARIES "rt")

--- a/recipes/folly/all/patches/0014-find-librt.patch
+++ b/recipes/folly/all/patches/0014-find-librt.patch
@@ -1,0 +1,18 @@
+diff --git a/CMake/FollyConfigChecks.cmake b/CMake/FollyConfigChecks.cmake
+index 6b8b308c7..908d72d51 100644
+--- a/CMake/FollyConfigChecks.cmake
++++ b/CMake/FollyConfigChecks.cmake
+@@ -83,6 +83,13 @@ string(REGEX REPLACE
+   CMAKE_REQUIRED_FLAGS
+   "${CMAKE_REQUIRED_FLAGS}")
+ 
++if (UNIX)
++  find_library(LIBRT rt)
++  if (LIBRT)
++    set(CMAKE_REQUIRED_LIBRARIES "rt")
++  endif()
++endif()
++
+ check_symbol_exists(pthread_atfork pthread.h FOLLY_HAVE_PTHREAD_ATFORK)
+ 
+ # Unfortunately check_symbol_exists() does not work for memrchr():

--- a/recipes/folly/all/patches/0015-benchmark-format-macros.patch
+++ b/recipes/folly/all/patches/0015-benchmark-format-macros.patch
@@ -1,0 +1,15 @@
+diff --git a/folly/Benchmark.cpp b/folly/Benchmark.cpp
+index 389ee46a1..390b7674b 100644
+--- a/folly/Benchmark.cpp
++++ b/folly/Benchmark.cpp
+@@ -16,6 +16,10 @@
+ 
+ // @author Andrei Alexandrescu (andrei.alexandrescu@fb.com)
+ 
++#ifndef __STDC_FORMAT_MACROS
++#define __STDC_FORMAT_MACROS 1
++#endif
++
+ #include <folly/Benchmark.h>
+ 
+ #include <algorithm>

